api/kiveapi/dataset.py

```diff
@@ -42,8 +42,9 @@ def download(self, handle):
 
         :param handle: A file handle
         """
-
-        for block in self._request_download().iter_content(1024):
+        blocks = self._request_download().iter_content(chunk_size=1024,
+                                                       decode_unicode=True)
```

Comment:
Does this work for downloading binary files like zip files? I assumed that encoding wouldn't matter if the source and destination were both in binary mode.

Reply:
I added this change because the API client code was choking on a bytes object. I suspect this is just because the existing example predates our Python 3 migration, so it was trying to put a bytes into a string buffer.

With this argument, the response if the response suggests that it should be decoded (it's run through this helper function in `requests`: https://github.com/psf/requests/blob/8149e9fe54c36951290f198e90d83c8a0498289c/requests/utils.py#L501).

I uploaded and downloaded both zip and text files with no problems.

I suppose it's *possible* that the other change in this commit is somehow responsible for the error in the client, but I tried hitting the API with and without it, and got byte-for-byte and header-for-header identical responses for both binary and text files, so I don't think that's the reason.